### PR TITLE
base64: use Rust base64 encoder - v2

### PR DESCRIPTION
**Pull request by jasonish**
Previous PR: https://github.com/OISF/suricata/pull/6227

Changes from last PR:
- Replace our C Base64Encode with a Rust wrapper.

Add methods jb_set_base64 and jb_append_base64 to JsonBuilder to set/append byte array values as base64. This avoids the intermediate buffer that we currently use, and the Rust base64 encode we use can encode directly to the underlying JsonBuilder buffer (internally it appears to use a 1024 byte intermediate buffer that is flushed to the output string when full).

Finally convert all occurrences of Base64Encode directly before using it in JsonBuilder with these new methods.

### CODE REVIEW

**Comment by suricata-qa** [comment]
Information: QA ran without warnings.

Pipeline 4051

File: src/log-tlsstore.c [hunk 1]
```diff
@@ -123,7 +122,7 @@ static void LogTlsLogPem(LogTlsStoreLogThread *aft, const Packet *p, SSLState *s
     }
 
     TAILQ_FOREACH(cert, &state->server_connp.certs, next) {
-        pemlen = BASE64_BUFFER_SIZE(cert->cert_len);
+        pemlen = Base64EncodeBufferSize(cert->cert_len);
```
Comment (victorjulien) [inline src/log-tlsstore.c:125]: I'm seeing a difference in the stored pem files. I'll share my example offline

**Review by victorjulien — CHANGES_REQUESTED** [review]
TLS log issue (shared privately)

**Comment by jasonish** [comment]
Issue should be fixed now. New PR: https://github.com/OISF/suricata/pull/6346

**Pull request closed without merging**